src/ZeroLog/LogEvent.cs

```diff
@@ -284,15 +302,143 @@ public void WriteToStringBuffer(StringBuffer stringBuffer)
+
+            // Save a pointer to the key for later when we append the Key/Value JSON.
+            keyValuePointers[keyValueCount++] = new IntPtr(dataPointer);
+
+            // Skip the key.
+            SkipCurrentArgument(ref dataPointer);
+
+            // Skip the value.
+            SkipCurrentArgument(ref dataPointer);
+
+            return true;
+        }
+
+        private static void SkipCurrentArgument(ref byte* dataPointer)
+        {
+            var argumentType = (ArgumentType)(*dataPointer & ArgumentTypeMask.ArgumentType);
+            dataPointer += sizeof(ArgumentType);
+            switch (argumentType)
+            {
+                case ArgumentType.String:
+                case ArgumentType.KeyString:
+                case ArgumentType.Byte:
+                    dataPointer += sizeof(byte);
+                    break;
+
+                case ArgumentType.AsciiString:
+                    var length = *(int*)dataPointer;
+                    dataPointer += sizeof(int) + length;
+                    break;
+
+                case ArgumentType.Boolean:
+                    dataPointer += sizeof(bool);
+                    break;
+
+                case ArgumentType.Char:
+                    dataPointer += sizeof(char);
+                    break;
+
+                case ArgumentType.Int16:
+                    dataPointer += sizeof(short);
+                    break;
+
+                case ArgumentType.Int32:
+                    dataPointer += sizeof(int);
+                    break;
+
+                case ArgumentType.Int64:
+                    dataPointer += sizeof(long);
+                    break;
+
+                case ArgumentType.Single:
+                    dataPointer += sizeof(float);
+                    break;
+
+                case ArgumentType.Double:
+                    dataPointer += sizeof(double);
+                    break;
+
+                case ArgumentType.Decimal:
+                    dataPointer += sizeof(decimal);
+                    break;
+
+                case ArgumentType.Guid:
+                    dataPointer += sizeof(Guid);
+                    break;
+
+                case ArgumentType.DateTime:
+                    dataPointer += sizeof(DateTime);
+                    break;
+
+                case ArgumentType.TimeSpan:
+                    dataPointer += sizeof(TimeSpan);
+                    break;
+
+                case ArgumentType.Enum:
+                    dataPointer += sizeof(EnumArg);
+                    break;
+
+                case ArgumentType.Null:
+                    break;
+                case ArgumentType.FormatString:
+                case ArgumentType.Unmanaged:
+                    throw new NotSupportedException($"Type is not supported {argumentType}");
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private void WriteJsonToStringBuffer(StringBuffer stringBuffer)
```

Comment:
I wouldn't put this method (and the associated ones) in this file. I know there's already `WriteToStringBufferUnformatted` in there, but that's also bad. Please try to put the JSON stuff in a separate file.